### ISSUE TITLE
Implement/improve validation of StructureDefinition resources

### ISSUE THREAD

**Issue by wmrutten** [ewoutkramer/fhir-net-api#856]
The validator implementation has been focused on validating a Resource instance against a profile.
However a profile is also a resource instance that can be validated against it's base profile c.q. StructureDefinition and ElementDefinition. A validator should then report all constraints that are invalid according to FHIR rules (e.g. relaxing instead of constraining).
Note: Simplifier already demonstrates that the validator is capable of validating StructureDefinitions, using the standard validation logic. However we can probably improve the quality and completeness of the report by implementing custom warnings specifically targeting StructureDefinition. This ticket is intended as a reminder.

**Comment by ewoutkramer** [comment]
In fact, this is probably an infinite task - of course we run all invariants and structural checks we can on StructureDefinition, but there are numerous checks that require hand-written code. E.g. the fact that you cannot have a differential that constrains a cardinality to a wider range than the base.

Or a differential that changes isModifier. etc. etc.

**Comment by wmrutten** [comment]
Exactly, it would be quite useful for profile authors if our validator could detect and report invalid constraints.
But i'm not expecting a complete implementation any time soon ;p

**Comment by cknaap** [comment]
Also mentioned as 'error' in Vonk: https://chat.fhir.org/#narrow/stream/179166-implementers/topic/Composition.20subject.200.2E.2E1/near/186679055

**Comment by ewoutkramer** [comment]
Another example of a rule we could add manually: #1392.

**Comment by ewoutkramer** [comment]
And also #1256.

**Comment by ewoutkramer** [comment]
More rules in #1261.

**Comment by ewoutkramer** [comment]
And #1254
